### ISSUE TITLE
Issues with virial radius computation

### ISSUE THREAD

**Issue by lluciesmith** [pynbody/pynbody#560]
Hi,

I am having issues with computing the virial radius of a halo. Here is an example of my code:
```
f = pynbody.load("snapshot")
f.physical_units()
h = f.halos()
pynbody.analysis.halo.center(h[halo_id], vel=False, wrap=True)
rvir = pynbody.analysis.halo.virial_radius(h[halo_id], overden=200)
```
This code does not return an error but the value it outputs is wrong. It is about 2x larger than what it should be. I was able to run the same code on a different machine for the same simulation and I found that there the virial radius function returns the correct value, which also agrees with the virial radius value in the halo properties. In trying to debug this, I found that it is line 171 of pynbody/pynbody/analysis/halo.py that returns two different values in the two different machines.

I ran the nosetests, which do in fact also return errors in the virial radius test. I will attach below the full error messages from the nosetests.

Have you encountered this error before and would you know how to fix it? I imagine it has something to do with compiling pynbody, if it works on one machine but does not work on another. Thank you very much for any help!

Luisa

```
(py36) Luisas-MacBook-Air:nose lls$ nosetests
/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/ramses.py:508: UserWarning: No header file found -- no particle block information available
  "No header file found -- no particle block information available")
...../Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/halo/ahf.py:149: UserWarning: Unable to write AHF_fpos file; performance will be reduced. Pass write_fpos=False to halo constructor to suppress this message.
  warnings.warn("Unable to write AHF_fpos file; performance will be reduced. Pass write_fpos=False to halo constructor to suppress this message.")
................./Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/simdict.py:92: RuntimeWarning: Assuming default value for property 'h'=6.78e-01
  name, value), RuntimeWarning)
/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/simdict.py:92: RuntimeWarning: Assuming default value for property 'omegaM0'=3.09e-01
  name, value), RuntimeWarning)
/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/simdict.py:92: RuntimeWarning: Assuming default value for property 'omegaL0'=6.91e-01
  name, value), RuntimeWarning)
./Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/analysis/cosmology.py:179: RuntimeWarning: invalid value encountered in log
  log_a_input = np.log(1./(1.+z))
/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/scipy/interpolate/interpolate.py:687: RuntimeWarning: invalid value encountered in less
  below_bounds = x_new < self.x[0]
/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/scipy/interpolate/interpolate.py:688: RuntimeWarning: invalid value encountered in greater
  above_bounds = x_new > self.x[-1]
................./Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/gadget.py:464: RuntimeWarning: Run out of block names in the config file. Using fallbacks: UNK*
  "Run out of block names in the config file. Using fallbacks: UNK*", RuntimeWarning)
./Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/gadget.py:1163: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  if np.issubdtype(data.dtype, float):
./Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/gadget.py:690: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  if np.issubdtype(dtype, float):
/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/gadget.py:692: FutureWarning: Conversion of the second argument of issubdtype from `int` to `np.signedinteger` is deprecated. In future, it will be treated as `np.int64 == np.dtype(int).type`.
  if np.issubdtype(dtype, int):
........./Users/lls/Software/pynbody/nose/gadgethdf_test.py:37: H5pyDeprecationWarning: The default file mode will change to 'r' (read-only) in h5py 3.0. To suppress this warning, pass the mode you need to h5py.File(), or set the global default h5.get_config().default_file_mode, or set the environment variable H5PY_DEFAULT_READONLY=1. Available modes are: 'r', 'r+', 'w', 'w-'/'x', 'a'. See the docs for details.
  f = h5py.File(dest)
../Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/halo/subfindhdf.py:154: H5pyDeprecationWarning: dataset.value has been deprecated. Use dataset[()] instead.
  props[property_key] = np.append(props[property_key], h[hdf_key][property_key].value)
./Users/lls/Software/pynbody/nose/pyread_gadget_hdf5.py:324: H5pyDeprecationWarning: dataset.value has been deprecated. Use dataset[()] instead.
  tmpset = pd.DataFrame(fin[global_var_name].value) ## Let HDF5 metadata dictate array format
/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/halo/subfindhdf.py:154: H5pyDeprecationWarning: dataset.value has been deprecated. Use dataset[()] instead.
  props[property_key] = np.append(props[property_key], h[hdf_key][property_key].value)
.........../Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/array.py:865: FutureWarning: Using a non-tuple sequence for multidimensional indexing is deprecated; use `arr[tuple(seq)]` instead of `arr[seq]`. In the future this will be interpreted as an array index, `arr[np.array(seq)]`, which will result either in an error or a different result.
  return self.base[self._reexpress_index(item)]
pynbody.analysis.profile : Profile v_circ -- this routine assumes the disk is in the x-y plane
F.FFF.FF./Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/analysis/hmf.py:806: UserWarning: Halo finder masses not provided. Calculating them (might take a while...)
  warnings.warn("Halo finder masses not provided. Calculating them (might take a while...)")
............../Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/ramses.py:688: UserWarning: Unexpected format in file testdata/ramses_new_format_partial_output_00001/sink_00001.csv -- sink data has not been loaded
  warnings.warn("Unexpected format in file %s -- sink data has not been loaded" % self._sink_filename())
....../Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/ramses.py:282: RuntimeWarning: More hydro variables (7) are in this RAMSES dump than are defined in config.ini (6)
  nvar_file, nvar), RuntimeWarning)
/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/ramses.py:282: RuntimeWarning: More hydro variables (7) are in this RAMSES dump than are defined in config.ini (6)
  nvar_file, nvar), RuntimeWarning)
/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/ramses.py:282: RuntimeWarning: More hydro variables (7) are in this RAMSES dump than are defined in config.ini (6)
  nvar_file, nvar), RuntimeWarning)
/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/ramses.py:282: RuntimeWarning: More hydro variables (7) are in this RAMSES dump than are defined in config.ini (6)
  nvar_file, nvar), RuntimeWarning)
.../Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/ramses.py:282: RuntimeWarning: More hydro variables (7) are in this RAMSES dump than are defined in config.ini (6)
  nvar_file, nvar), RuntimeWarning)
/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/ramses.py:282: RuntimeWarning: More hydro variables (7) are in this RAMSES dump than are defined in config.ini (6)
  nvar_file, nvar), RuntimeWarning)
/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/ramses.py:282: RuntimeWarning: More hydro variables (7) are in this RAMSES dump than are defined in config.ini (6)
  nvar_file, nvar), RuntimeWarning)
/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/ramses.py:282: RuntimeWarning: More hydro variables (7) are in this RAMSES dump than are defined in config.ini (6)
  nvar_file, nvar), RuntimeWarning)
../Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/ramses.py:761: UserWarning: Using field at offset 9 to distinguish stars. If this is wrong, try editing your config.ini, section [ramses], entry particle-distinguisher.
  warnings.warn("Using field %s to distinguish stars. If this is wrong, try editing your config.ini, section [ramses], entry particle-distinguisher."%pb_name)
.............................Exception in thread Thread-66:
Traceback (most recent call last):
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/sph/__init__.py", line 440, in bridge
    X = fn(*args[1:], **kwargs)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/sph/__init__.py", line 422, in render_fn
    base = fn(*args, **kwargs)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/sph/__init__.py", line 572, in _render_image
    snap_proxy[arname] = snap[arname]
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/__init__.py", line 264, in __getitem__
    return self._get_array_with_lazy_actions(i)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/__init__.py", line 357, in _get_array_with_lazy_actions
    self.__derive_if_required(name)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/__init__.py", line 372, in __derive_if_required
    self._derive_array(name)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/__init__.py", line 2016, in _derive_array
    self.base._derive_array(array_name, self._unifamily)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/__init__.py", line 1502, in _derive_array
    result = fn(self[fam])
  File "/Users/lls/Software/pynbody/nose/sph_image_test.py", line 92, in intentional_circular_reference
    return sim['intentional_circular_reference']
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/__init__.py", line 264, in __getitem__
    return self._get_array_with_lazy_actions(i)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/__init__.py", line 350, in _get_array_with_lazy_actions
    with self._dependency_tracker.calculating(name):
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/dependencytracker.py", line 16, in __enter__
    raise DependencyError("Circular dependency")
pynbody.dependencytracker.DependencyError: Circular dependency

Exception in thread Thread-65:
Traceback (most recent call last):
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/sph/__init__.py", line 440, in bridge
    X = fn(*args[1:], **kwargs)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/sph/__init__.py", line 422, in render_fn
    base = fn(*args, **kwargs)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/sph/__init__.py", line 572, in _render_image
    snap_proxy[arname] = snap[arname]
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/__init__.py", line 264, in __getitem__
    return self._get_array_with_lazy_actions(i)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/__init__.py", line 357, in _get_array_with_lazy_actions
    self.__derive_if_required(name)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/__init__.py", line 372, in __derive_if_required
    self._derive_array(name)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/__init__.py", line 2016, in _derive_array
    self.base._derive_array(array_name, self._unifamily)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/__init__.py", line 1502, in _derive_array
    result = fn(self[fam])
  File "/Users/lls/Software/pynbody/nose/sph_image_test.py", line 92, in intentional_circular_reference
    return sim['intentional_circular_reference']
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/__init__.py", line 264, in __getitem__
    return self._get_array_with_lazy_actions(i)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/__init__.py", line 350, in _get_array_with_lazy_actions
    with self._dependency_tracker.calculating(name):
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/dependencytracker.py", line 16, in __enter__
    raise DependencyError("Circular dependency")
pynbody.dependencytracker.DependencyError: Circular dependency
Exception in thread Thread-67:
Traceback (most recent call last):
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/sph/__init__.py", line 440, in bridge
    X = fn(*args[1:], **kwargs)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/sph/__init__.py", line 422, in render_fn
    base = fn(*args, **kwargs)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/sph/__init__.py", line 572, in _render_image
    snap_proxy[arname] = snap[arname]
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/__init__.py", line 264, in __getitem__
    return self._get_array_with_lazy_actions(i)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/__init__.py", line 357, in _get_array_with_lazy_actions
    self.__derive_if_required(name)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/__init__.py", line 372, in __derive_if_required
    self._derive_array(name)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/__init__.py", line 2016, in _derive_array
    self.base._derive_array(array_name, self._unifamily)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/__init__.py", line 1502, in _derive_array
    result = fn(self[fam])
  File "/Users/lls/Software/pynbody/nose/sph_image_test.py", line 92, in intentional_circular_reference
    return sim['intentional_circular_reference']
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/__init__.py", line 264, in __getitem__
    return self._get_array_with_lazy_actions(i)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/__init__.py", line 350, in _get_array_with_lazy_actions
    with self._dependency_tracker.calculating(name):
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/dependencytracker.py", line 16, in __enter__
    raise DependencyError("Circular dependency")
pynbody.dependencytracker.DependencyError: Circular dependency


Exception in thread Thread-64:
Traceback (most recent call last):
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/sph/__init__.py", line 440, in bridge
    X = fn(*args[1:], **kwargs)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/sph/__init__.py", line 422, in render_fn
    base = fn(*args, **kwargs)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/sph/__init__.py", line 572, in _render_image
    snap_proxy[arname] = snap[arname]
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/__init__.py", line 264, in __getitem__
    return self._get_array_with_lazy_actions(i)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/__init__.py", line 357, in _get_array_with_lazy_actions
    self.__derive_if_required(name)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/__init__.py", line 372, in __derive_if_required
    self._derive_array(name)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/__init__.py", line 2016, in _derive_array
    self.base._derive_array(array_name, self._unifamily)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/__init__.py", line 1502, in _derive_array
    result = fn(self[fam])
  File "/Users/lls/Software/pynbody/nose/sph_image_test.py", line 92, in intentional_circular_reference
    return sim['intentional_circular_reference']
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/__init__.py", line 264, in __getitem__
    return self._get_array_with_lazy_actions(i)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/__init__.py", line 350, in _get_array_with_lazy_actions
    with self._dependency_tracker.calculating(name):
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/dependencytracker.py", line 16, in __enter__
    raise DependencyError("Circular dependency")
pynbody.dependencytracker.DependencyError: Circular dependency

..........pynbody.analysis.profile : Profile pot -- this routine assumes the disk is in the x-y plane
.pynbody.analysis.profile : Profile pot -- this routine assumes the disk is in the x-y plane
........./Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/pynbody/snapshot/tipsy.py:1578: RuntimeWarning: Paramfile suggests time is cosmological, but header values are not sensible in this context.
  "Paramfile suggests time is cosmological, but header values are not sensible in this context.", RuntimeWarning)
............................F..
======================================================================
FAIL: gravity_test.test_gravity
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/lls/Software/pynbody/nose/gravity_test.py", line 29, in test_gravity
    npt.assert_allclose(v_circ, v_circ_correct,atol=1e-5)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/numpy/testing/_private/utils.py", line 1533, in assert_allclose
    verbose=verbose, header=header, equal_nan=equal_nan)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/numpy/testing/_private/utils.py", line 846, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Not equal to tolerance rtol=1e-07, atol=1e-05

Mismatched elements: 50 / 50 (100%)
Max absolute difference: 331.25281372
Max relative difference: 1.00001606
 x: SimArray([ 71.619741, 126.875793, 154.919288, 177.234475, 199.673627,
          256.433703, 361.467643, 374.88502 , 384.897137, 457.238933,
          478.81388 , 469.297267, 511.12285 , 505.537492, 523.780709,...
 y: array([ 57.196343, 103.043315, 132.124116, 155.652758, 175.528504,
       193.261025, 209.524186, 224.379998, 237.986664, 250.735445,
       262.769121, 273.847541, 283.903991, 293.118261, 301.351452,...
-------------------- >> begin captured logging << --------------------
pynbody.analysis.profile: WARNING: Profile v_circ -- this routine assumes the disk is in the x-y plane
--------------------- >> end captured logging << ---------------------

======================================================================
FAIL: gravity_test.test_eps_retrieval_str
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/lls/Software/pynbody/nose/gravity_test.py", line 50, in test_eps_retrieval_str
    npt.assert_allclose(f['phi'][:10], true_phi_10)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/numpy/testing/_private/utils.py", line 1533, in assert_allclose
    verbose=verbose, header=header, equal_nan=equal_nan)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/numpy/testing/_private/utils.py", line 846, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Not equal to tolerance rtol=1e-07, atol=0

Mismatched elements: 10 / 10 (100%)
Max absolute difference: 0.12303491
Max relative difference: 1.76358358
 x: SimArray([-0.16027 , -0.193906, -0.156042, -0.186238, -0.153566,
          -0.079105, -0.153086, -0.128889, -0.167301, -0.140598],
         dtype=float32)
 y: array([-0.066966, -0.070871, -0.070492, -0.06739 , -0.067484, -0.069525,
       -0.068039, -0.067983, -0.07278 , -0.071891])

======================================================================
FAIL: gravity_test.test_eps_retrieval_unit
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/lls/Software/pynbody/nose/gravity_test.py", line 60, in test_eps_retrieval_unit
    npt.assert_allclose(f['phi'][:10], true_phi_10)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/numpy/testing/_private/utils.py", line 1533, in assert_allclose
    verbose=verbose, header=header, equal_nan=equal_nan)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/numpy/testing/_private/utils.py", line 846, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Not equal to tolerance rtol=1e-07, atol=0

Mismatched elements: 10 / 10 (100%)
Max absolute difference: 0.18862942
Max relative difference: 2.59178801
 x: SimArray([-0.115887, -0.122962, -0.183271, -0.148383, -0.184503,
          -0.183351, -0.191217, -0.212269, -0.261409, -0.156443],
         dtype=float32)
 y: array([-0.066966, -0.070871, -0.070492, -0.06739 , -0.067484, -0.069525,
       -0.068039, -0.067983, -0.07278 , -0.071891])

======================================================================
FAIL: gravity_test.test_eps_retrieval_number
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/lls/Software/pynbody/nose/gravity_test.py", line 70, in test_eps_retrieval_number
    npt.assert_allclose(f['phi'][:10], true_phi_10)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/numpy/testing/_private/utils.py", line 1533, in assert_allclose
    verbose=verbose, header=header, equal_nan=equal_nan)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/numpy/testing/_private/utils.py", line 846, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Not equal to tolerance rtol=1e-07, atol=0

Mismatched elements: 10 / 10 (100%)
Max absolute difference: 0.12244652
Max relative difference: 1.72772661
 x: SimArray([-0.158451, -0.193318, -0.159647, -0.164957, -0.130715,
          -0.098073, -0.156272, -0.101791, -0.16459 , -0.142858],
         dtype=float32)
 y: array([-0.066966, -0.070871, -0.070492, -0.06739 , -0.067484, -0.069525,
       -0.068039, -0.067983, -0.07278 , -0.071891])

======================================================================
FAIL: halotools_test.test_align
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/lls/Software/pynbody/nose/halotools_test.py", line 30, in test_align
    [-0.010026,  0.002441, -0.04465 ]], atol=1e-5)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/numpy/testing/_private/utils.py", line 1533, in assert_allclose
    verbose=verbose, header=header, equal_nan=equal_nan)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/numpy/testing/_private/utils.py", line 846, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Not equal to tolerance rtol=1e-07, atol=1e-05

Mismatched elements: 4 / 6 (66.7%)
Max absolute difference: 2.07612563e-05
Max relative difference: 0.00882118
 x: SimArray([[-0.010698, -0.001517, -0.044787],
          [-0.010005,  0.002427, -0.044655]])
 y: array([[-0.010718, -0.001504, -0.044783],
       [-0.010026,  0.002441, -0.04465 ]])

======================================================================
FAIL: halotools_test.test_virialradius
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/lls/Software/pynbody/nose/halotools_test.py", line 42, in test_virialradius
    np.testing.assert_allclose(vrad, 0.005946911872, atol=1.e-5)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/numpy/testing/_private/utils.py", line 1533, in assert_allclose
    verbose=verbose, header=header, equal_nan=equal_nan)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/numpy/testing/_private/utils.py", line 846, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Not equal to tolerance rtol=1e-07, atol=1e-05

Mismatched elements: 1 / 1 (100%)
Max absolute difference: 0.0037955
Max relative difference: 0.63822963
 x: SimArray(0.009742)
 y: array(0.005947)
-------------------- >> begin captured stdout << ---------------------
time= 0.5607659816741943

--------------------- >> end captured stdout << ----------------------

======================================================================
FAIL: utils_test.test_openmp_summations
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/lls/Software/pynbody/nose/utils_test.py", line 134, in test_openmp_summations
    npt.assert_allclose(sum_a_ne,sum_a)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/numpy/testing/_private/utils.py", line 1533, in assert_allclose
    verbose=verbose, header=header, equal_nan=equal_nan)
  File "/Users/lls/anaconda/envs/py36/lib/python3.6/site-packages/numpy/testing/_private/utils.py", line 846, in assert_array_compare
    raise AssertionError(msg)
AssertionError: 
Not equal to tolerance rtol=1e-07, atol=0

Mismatched elements: 1 / 1 (100%)
Max absolute difference: 770.62013981
Max relative difference: 3.
 x: array(-1027.49352)
 y: array(-256.87338)
-------------------- >> begin captured stdout << ---------------------
NP: 1620.7 NE: 456.2
  vals -1027.5 -256.9

--------------------- >> end captured stdout << ----------------------

----------------------------------------------------------------------
Ran 179 tests in 127.809s

FAILED (failures=7)
```

**Comment by apontzen** [comment]
Thanks for the report. I strongly suspect this is not a problem with pynbody per se but rather one of the underlying libraries. In particular with anaconda on mac os I have often had problems with bizarre errors within numpy calculations. Disabling MKL acceleration seems to solve these for me, and there are instructions here:

https://docs.anaconda.com/mkl-optimizations/#uninstalling-mkl

Could you give this a go and let us know if it helps?

**Comment by lluciesmith** [comment]
@apontzen thank you for the suggestion. Unfortunately disabling MKL did not solve it. 

I was able to solve it in a different way though. The problem had to do with the installation, not a bug in the actual pynbody library. I was able to fix it by re-installing the relevant underlying packages in a new anaconda environment (python 3.5), compiling pynbody with a gcc compiler installed from anaconda, disabling the macosx backend in matplotlib and fixing a few potential conflicts between homebrew and anaconda. I am not sure exactly which of the underlying libraries was causing the initial issue though.
